src/libraries/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -221,13 +221,13 @@ public static IEnumerable<string> EnumerateFileSystemEntries(string path, string
         public static IEnumerable<string> EnumerateFileSystemEntries(string path, string searchPattern, EnumerationOptions enumerationOptions)
             => InternalEnumeratePaths(path, searchPattern, SearchTarget.Both, enumerationOptions);
 
-        public static string GetDirectoryRoot(string path)
+        public static string? GetDirectoryRoot(string path)
         {
             if (path == null)
                 throw new ArgumentNullException(nameof(path));
 
             string fullPath = Path.GetFullPath(path);
-            string root = Path.GetPathRoot(fullPath);
+            string? root = Path.GetPathRoot(fullPath);
```

Comment:
Doc saying it `Path.GetPathRoot(...)` returns null if fullPath is null or is effectively empty. The fullPath will not be null here but I was not sure if it can be empty, so made it nullable and return nullable. But seems it wouldn't be empty as it is throwing if empty path given, so will bang here and will return non null

Reply:
Yeah, it makes sense to not return nullable if we throw when empty.